python/ray/external_storage.py

```diff
@@ -1,8 +1,67 @@
 import abc
 import os
+import urllib
+from collections import namedtuple
 from typing import List
 
 import ray
+from ray.ray_constants import DEFAULT_OBJECT_PREFIX
+from ray._raylet import ObjectRef
+
+
+def create_url_with_offsets(*, url: str, offsets_first: int,
+                            offsets_end: int) -> str:
+    """Methods to create an url with offsets.
+
+    When ray spills objects, it fuses multiple objects
+    into one file to optimize the performance. That says, each object
+    needs to keep tracking of its own special url to store metadata.
+
+    This method creates an url_with_offset, which is used internally
+    by Ray.
+
+    Created url_with_offset can be passed to the self._get_base_url method
+    to parse the filename used to store files.
+
+    Example) file://path/to/file?offsets_first=""&offsets_end=""
+
+    Args:
+        url(str): url to the object stored in the external storage.
+        offsets_first(int): Offsets from the beginning of the file to
```

Comment:
Can we use offset and size instead of first/end? It's easier to understand (more common) and doesn't require any explanation.